File: 240415/특정 조건의 수 뽑기/draw-number-of-specific-conditions.kt
fun main() {
    val n = readLine()!!.toInt()

    var closest = 200
    var above = -1

    val numbers = readLine()!!.split(" ").map { it.toInt() }

    for(num in numbers) {
        if(Math.abs(100 - num) < Math.abs(100 - closest)) {
            closest = num
        }

        if(num >= 100 && (above == -1 || num < above)) {
            above = num
        }
    }

    println("$closest ${if (above == -1) -1 else above}")
}

File: 240415/특정 조건의 수 뽑기/draw-number-of-specific-conditions.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int n;
    scanf("%d", &n);
    
    int closest = 200;  
    int above = -1;     

    for (int i = 0; i < n; i++) {
        int num;
        scanf("%d", &num);
        
        if (abs(100 - num) < abs(100 - closest)) {
            closest = num;
        } else if (num >= 100 && (above == -1 || num < above)) {
            above = num;
        }
    }

    printf("%d %d\n", closest, above);
    
    return 0;
}